syncr_backend/util/drop_util.py

```diff
@@ -181,6 +195,73 @@ def get_drop_metadata(
     return metadata
 
 
+def simple_get_drop_metadata(drop_id: bytes) -> DropMetadata:
+    """
+    Get drop_metadata object from just drop_id
+    :param drop_id:
+    :return: A drop_metadata object
+    """
+    peers = get_drop_peers(drop_id)
+    drop_metadata = get_drop_metadata(drop_id, peers)
+
+    return drop_metadata
+
+
+def get_owned_drops_metadata() -> List[DropMetadata]:
+    """
+    Get list of metadata objects for owned drops (primary and secondary)
+    :return: list of metadata objects this node owns
+    """
+    drops = list_drops()
+
+    # Get current nodes id
+    priv_key = node_init.load_private_key_from_disk()
+    node_id = crypto_util.node_id_from_public_key(priv_key.public_key())
+
+    owned_drops = []
+
+    for drop_id in drops:
+        # Get drop_metadata object for drop
+        md = simple_get_drop_metadata(drop_id)
+        if md.owner == node_id:
+            owned_drops.append(md)
+        else:
+            for owner in md.other_owners:
+                if owner == node_id:
+                    owned_drops.append(md)
+
+    return owned_drops
+
+
+def get_subscribed_drops_metadata() -> List[DropMetadata]:
+    """
+    Get list of metadata objects for subscribed drops
+    :return: list of metadata objects this node is subscribed to
+    """
+    drops = list_drops()
+
+    # Get current nodes id
+    priv_key = node_init.load_private_key_from_disk()
+    node_id = crypto_util.node_id_from_public_key(priv_key.public_key())
+
+    subscribed_drops = []
+
+    # Subscribed drops are those on the disk that this node does not own
+    for drop_id in drops:
+        # Get drop_metadata object for drop
+        md = simple_get_drop_metadata(drop_id)
+        if md.owner == node_id:
+            continue
+        else:
+            secondary_owner = False
+            if node_id in md.other_owners:
+                continue
+            else:
```

Comment:
this doesn't have to be behind a else with the continue

Reply:
simplified 👍